src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -191,7 +191,7 @@ protected void processFiltered(File file, FileText fileText) throws CheckstyleEx
             }
             else {
                 final SortedSet<LocalizedMessage> filteredMessages =
-                    getFilteredMessages(file.getPath(), contents, rootAST);
+                    getFilteredMessages(file.getAbsolutePath(), contents, rootAST);
```

Comment:
it there any way to make test on this ?
if not possible to do in usual way, I am ok with reflection.

Reply:
It only happens when a file is given and no directory path at all, or the partial directory path given is not part of the suppression directory given.

In our case I can't give "src" path and suppression text would have to be something that comes before `src/`.

The only way to do this in our setup is verify the `/` for the directory path appears right before `src`.

This is done.